clients/js/src/embeddings/VoyageAIEmbeddingFunction.ts

```diff
@@ -0,0 +1,87 @@
+import { IEmbeddingFunction } from "./IEmbeddingFunction";
+
+export enum InputType {
+  DOCUMENT = "document",
+  QUERY = "query"
+}
+
+export class VoyageAIEmbeddingFunction implements IEmbeddingFunction {
+  private model_name: string;
+  private api_url: string;
+  private batch_size: number;
+  private truncation?: boolean;
+  private input_type?: InputType;
+  private headers: { [key: string]: string };
+
+  constructor({
+    voyageai_api_key,
+    model_name,
+    batch_size,
+    truncation,
+    input_type,
+  }: {
+    voyageai_api_key: string;
+    model_name: string;
+    batch_size?: number;
+    truncation?: boolean;
+    input_type?: InputType;
+  }) {
+    this.api_url = "https://api.voyageai.com/v1/embeddings";
+    this.headers = {
+      Authorization: `Bearer ${voyageai_api_key}`,
+      "Content-Type": "application/json",
+    };
+
+    this.model_name = model_name;
+    this.truncation = truncation;
+    this.input_type = input_type;
+    if (batch_size) {
+      this.batch_size = batch_size;
+    } else {
+      if (model_name in ["voyage-2", "voyage-02"]) {
```

Comment:
Yes, our API handles maximum 128 texts, so thats the max. batch size.
https://docs.voyageai.com/docs/embeddings

Reply:
Can you elaborate on the `72/7` number? If you're saying the max number of tokens per request wouldn't, then the calculation be:

`sum(len(input) for input in inputs)/4<max_tokens_per_req` (assuming 4 letters is the average token length)?

Or maybe I am missing something?

Reply:
> `sum(len(input) for input in inputs)/4<max_tokens_per_req` (assuming 4 letters is the average token length). 

This can still lead to error in corner cases since one token can have 16 characters at most.   

The ideal way is to include voyage ai tokenizers https://docs.voyageai.com/docs/tokenization and count the number of tokens in a batch. Assuming we don't want to introduce further dependency, 72 / 7 is the number we recommend and implemented in our partners' library.

Reply:
@Liuhong99, thanks for the explanation. Indeed, tokenizers are probably the best way to do it if they have native Python support. This has been a long-standing discussion: https://github.com/chroma-core/chroma/pull/1008. If `72/7` is the recommended way, then let's go with it.

Reply:
Sounds great, let go with 72 / 7. Thanks!